packages/base-map/src/index.tsx

```diff
@@ -0,0 +1,164 @@
+ * BaseMap. The user uses that control to turn overlays on or off. Only overlays
+ * with an id are added to the control.
+ */
+type Props = {
+  baseLayer?: string;
+  center?: [number, number];
+  children?: JSX.Element | JSX.Element[];
+  forceMaxHeight?: boolean;
+  maxZoom?: number;
+  onClick?: (evt: Event) => void;
+  // Unknown is used here because of a maplibre/mapbox issue with the true type, MapLayerMouseEvent
+  onContextMenu?: (e: unknown) => void;
+  onViewportChanged?: ({
+    latitude,
+    longitude,
+    zoom
+  }: {
+    latitude: number;
+    longitude: number;
+    zoom: number;
+  }) => void;
+  passedRef?: React.Ref<MapRef>;
+  zoom?: number;
+};
+const BaseMap = ({
+  // These tiles are free to use, but not in production
+  baseLayer = "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
+  center,
+  children,
+  forceMaxHeight,
+  maxZoom,
+  onClick,
+  onContextMenu,
+  passedRef,
+  onViewportChanged,
+  zoom: initZoom = 12
+}: Props): JSX.Element => {
+  const [viewState, setViewState] = React.useState<{
+    latitude: number;
+    longitude: number;
+    zoom: number;
+    fitBoundsOptions?: Record<string, number | string | boolean>;
+  }>({
+    fitBoundsOptions: {
+      animate: true,
+      padding: 10,
+      duration: 100,
+      maxDuration: 300,
+      essential: false
+    },
+    latitude: center?.[0],
+    longitude: center?.[1],
+    zoom: initZoom
+  });
+
+  useEffect(() => {
+    callIfValid(onViewportChanged)(viewState);
+  }, [viewState]);
+
+  const toggleableLayers = Array.isArray(children)
+    ? children
+        .flat()
+        .filter(child => child?.props?.id !== undefined)
+        .map(child => {
+          const { visible, name, id } = child.props;
+          return { visible, name, id };
+        })
+    : [];
+
+  const [hiddenLayers, setHiddenLayers] = useState(
+    toggleableLayers.filter(layer => !layer?.visible).map(layer => layer.id)
+  );
+
+  return (
+    <MapProvider>
+      <Map
+        ref={passedRef}
+        id="mainMap"
+        latitude={viewState.latitude}
+        longitude={viewState.longitude}
+        mapLib={maplibregl}
+        mapStyle={baseLayer}
+        maxZoom={maxZoom}
+        onClick={onClick}
+        onContextMenu={onContextMenu}
+        onMove={evt => setViewState(evt.viewState)}
+        style={{
+          display: "block",
+          width: "100%",
+          height: forceMaxHeight ? "100vh" : "100%"
+        }}
+        zoom={viewState.zoom}
+      >
+        {toggleableLayers.length > 0 && (
+          // TODO: Mobile view
+          <Styled.LayerSelector id="filter-group" className="filter-group">
+            <div className="layers-list">
+              {toggleableLayers.map((layer: LayerProps, index: number) => {
+                return (
+                  <label htmlFor={layer.id} key={index}>
+                    <input
+                      onChange={() => {
```

Comment:
Might need to memoize this method.

Reply:
Wouldn't every `hiddenLayer` change trigger a re-memoization? Or what were you thinking the dependencies would be